netutils/esp8266/esp8266.c

```diff
@@ -2672,8 +2673,10 @@ int lesp_setsockopt(int sockfd, int level, int option,
           case SO_RCVTIMEO:
               if (value_len == sizeof(struct timeval))
                 {
-                  sock->rcv_timeo.tv_sec = ((struct timeval *)(value))->tv_sec;
-                  sock->rcv_timeo.tv_nsec = ((struct timeval *)(value))->tv_usec;
+                  sock->rcv_timeo.tv_sec = ((struct timeval *)
```

Comment:
maybe adding a temp variable can be an option here instead.
```
struct timeval *tv = (struct timeval *)value;
sock->rcv_timeo.tv_sec = tv->tv_sec;
sock->rcv_timeo.tv_nsec = tv->tv_usec;
```

Reply:
this is not a nxstyle change!

Reply:
Ok. Let's keep it as is. Np.